### ISSUE TITLE
[AUDIT] [SPARK-49653][SQL] Single join for correlated scalar subqueries

### ISSUE THREAD

**Issue by amahussein** [NVIDIA/spark-rapids#11690]
https://github.com/apache/spark/commit/80d6651cf6a

This Spark PR introduces a new join type `LeftSingle` that triggered changes in the query execution 

- `sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledJoin.scala`
- `sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala`
- `sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastNestedLoopJoinExec.scala`
- `sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/PlanAdaptiveSubqueries.scala`

The PR adds single join implementation to support correlated scalar subqueries and introduces a new config `spark.sql.optimizer.scalarSubqueryUseSingleJoin`

**Comment by revans2** [comment]
I don't think that the link above is for this issue. The PR is https://github.com/apache/spark/pull/48145 which appears to correspond to https://github.com/apache/spark/commit/3c81f076ab9c72514cfc8372edd16e6da7c151d6 instead

**Comment by nartal1** [comment]
Moving the priority of this to P1 
cc: @revans2
